components/butterfly/src/rumor.rs

```diff
@@ -119,324 +115,385 @@ impl<'a, T: Rumor> From<&'a T> for RumorKey {
     fn from(rumor: &'a T) -> RumorKey { RumorKey::new(rumor.kind(), rumor.id(), rumor.key()) }
 }
 
-/// Storage for Rumors. It takes a rumor and stores it according to the member that produced it,
-/// and the service group it is related to.
-///
-/// Generic over the type of rumor it stores.
-#[derive(Debug, Clone)]
-pub struct RumorStore<T: Rumor> {
-    pub list:       Arc<RwLock<HashMap<String, HashMap<String, T>>>>,
-    update_counter: Arc<AtomicUsize>,
-}
-
-impl<T> Default for RumorStore<T> where T: Rumor
-{
-    fn default() -> RumorStore<T> {
-        RumorStore { list:           Arc::new(RwLock::new(HashMap::new())),
-                     update_counter: Arc::new(AtomicUsize::new(0)), }
-    }
-}
-
-impl<T> Deref for RumorStore<T> where T: Rumor
-{
-    type Target = RwLock<HashMap<String, HashMap<String, T>>>;
+type ServiceGroupName = String;
+type MemberId = String;
+type ServiceGroupRumorMap<T> = HashMap<MemberId, T>;
+type RumorMap<T> = HashMap<ServiceGroupName, ServiceGroupRumorMap<T>>;
```

Comment:
As I said in a comment on a previous commit, I think these names are misleading.  Something like the following would be more accurate:

```rust
type RumorKey = String;
type RumorId = String;
type RumorKeyRumorMap<T> = HashMap<RumorId, T>;
type RumorMap<T> = HashMap<RumorKey, RumorKeyRumorMap>;
```

I'm not stuck on those particular names, _per se_, only something that illustrates that the top level key is the result of calling `Rumor::key()`, the actual data of which can vary from rumor to rumor, and the second level key inside the nested map is the result of calling `Rumor::id()` (again, the actual data of which can vary).

Reply:
Agreed. I'm not wedded to this naming; I just wanted some simplification and indication of intent over `Arc<RwLock<HashMap<String, HashMap<String, T>>>>`

Reply:
We can't use `RumorKey` since that would conflict with the existing type (and putting the alias in a submodule would just be confusing), so I used the best names I could think of and integrated them into the `RumorKey` type as well. Let me know how you feel about https://github.com/habitat-sh/habitat/pull/6674/commits/0aa2082b0a7b326b092405a9fd3f973cac60dd49.

Reply:
Of course you're right about `RumorKey`.  I have no idea how I missed that.  I think your change looks great.

Reply:
Actually one more thing: 

It was a mistake to make `RumorStore::key()` return `RumorKeyKey`. But I realize I mad that mistake because `RumorStore::key()` is a misleading name for this method. I've switched it to use `to_string` in https://github.com/habitat-sh/habitat/pull/6674/commits/038a4729da21a1bab5dcafe0181539de2584f27a.